### PR TITLE
Expand textarea value/defaultValue/textContent tests

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/2766.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243428982)

# Firefox (nightly) #
Testing web-platform-tests at revision 2781fd7b3b841c377750c1f00acd2b09ae622c8b
Using browser at version BuildID 20170612100241; SourceStamp 27cad9749cddf68e11fdd4e5d73dad84a8f8cf23
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6256/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                              Subtest                                              | Results |                                                               Messages                                                              |
|---------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                   | OK      |                                                                                                                                     |
| `defaultValue and value are the empty string by default`                                          | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by setting textContent`                                      | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by setting nodeValue on a child text node`                   | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by setting data on a child text node`                        | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by textContent in combination with appending a text node`    | PASS    |                                                                                                                                     |
| `defaultValue and value reflect child text content, not textContent`                              | PASS    |                                                                                                                                     |
| `Setting defaultValue wipes out any children, including elements (just like setting textContent)` | PASS    |                                                                                                                                     |
| `defaultValue and value treat CRLF differently`                                                   | FAIL    | `assert_equals: The value property should normalize CRLF and CR to LF expected "foo\nbar\nbaz\nqux" but got "foo\r\nbar\rbaz\nqux"` |
| `value normalizes CRLF even spread over multiple text nodes`                                      | FAIL    | `assert_equals: The value property should normalize CRLF and CR to LF expected "foo\nbar\nbaz\nqux" but got "foo\r\nbar\rbaz\nqux"` |
| `tests for the value setter`                                                                      | PASS    |                                                                                                                                     |
| `tests for U+0000 NULL`                                                                           | PASS    |                                                                                                                                     |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/6256/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml</a></summary>


|                         Subtest                          | Results | Messages |
|----------------------------------------------------------|---------|----------|
|                                                          | OK      |          |
| `defaultValue and value include CDATASection Text nodes` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243428983)

# Sauce (safari) #
Testing web-platform-tests at revision 2781fd7b3b841c377750c1f00acd2b09ae622c8b
Using browser at version 10.0
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243428989)

# Chrome (unstable) #
Testing web-platform-tests at revision 2781fd7b3b841c377750c1f00acd2b09ae622c8b

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243428992)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 2781fd7b3b841c377750c1f00acd2b09ae622c8b

**Comment by zcorpan** [comment]
Can you also test inserting a `DocumentFragment` containing a `Text` or `CDATASection` node?

**Comment by domenic** [comment]
Good catch, done.

**Comment by domenic** [comment]
@bzbarsky or @tkent-google, could you review these? I believe @zcorpan is unavailable for a while.

File: html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tkent-google) [inline html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml]: Please rename this to value-defaultValue-textContext-xhtml.xhtml.
Basename-identical tests will cause troubles in WebKit and Blink.

**Review by tkent-google — CHANGES_REQUESTED** [review]

File: html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml (hunk 1 above)
Comment (domenic) [inline html/semantics/forms/the-textarea-element/value-defaultValue-textContent.xhtml]: Fixed, thanks!

**Review by tkent-google — APPROVED** [review]
lgtm

**Pull request merged**